src/jmclient/wallet.py

```diff
@@ -2740,30 +2800,41 @@ def _get_bip32_export_path(self, mixdepth=None, address_type=None):
         path = super()._get_bip32_export_path(mixdepth, address_type)
         return path
 
-    def _get_key_from_path(self, path):
+    def _get_key_from_path(self, path,
+            validate_cache: bool = False):
         raise WalletError("Cannot get a private key from a watch-only wallet")
 
-    def _get_keypair_from_path(self, path):
+    def _get_keypair_from_path(self, path,
+            validate_cache: bool = False):
         raise WalletError("Cannot get a private key from a watch-only wallet")
 
-    def _get_pubkey_from_path(self, path):
+    def _get_pubkey_from_path(self, path,
+            validate_cache: bool = False):
         if not self._is_my_bip32_path(path):
-            return super()._get_pubkey_from_path(path)
+            return super()._get_pubkey_from_path(path,
+                validate_cache=validate_cache)
         if self.is_timelocked_path(path):
             key_path = path[:-1]
             locktime = path[-1]
             cache = self._get_cache_for_path(key_path)
             pubkey = cache.get(b'P')
-            if pubkey is None:
-                pubkey = self._TIMELOCK_ENGINE.derive_bip32_privkey(
+            if pubkey is None or validate_cache:
+                new_pubkey = self._TIMELOCK_ENGINE.derive_bip32_privkey(
```

Comment:
i can't remember exactly what was going on here with the `pubkey` vs `privkey` naming in this class and its corresponding 'engine' (which we don't actually use yet), but it's of course entirely orthogonal to this PR